tests/commands/test_commands.py

```diff
@@ -409,3 +409,28 @@ def mock_write(tx, mode):
     resp = run_upsert_validator_new(args, b)
 
     assert b.get_transaction(resp)
+
+
+@pytest.mark.abci
+@pytest.mark.bdb
+def test_upsert_validator_approve(b, priv_validator_path, user_sk, validators):
```

Comment:
This test is leaving the test environment in a dirty state. Any test that happens after this one is executed get a wrong validator set, and hangs. We need a strategy for testing `upsert-validator` that can act on a real Tendermint instance, but restore it to it's original state when the test is done.

We should have a strategy session about this, since we're going to need this when we start testing Tendermint migrations, and so on.

Reply:
Fixed. The problem was some weird interaction between the `abci` and `bdb` decorators. Still don't know the root cause, but we don't seem to need the `bdb` marker for this test.